cmd/prow-job-dispatcher/main.go

```diff
@@ -61,9 +83,257 @@ func (o *options) complete(secrets *sets.String) error {
 		}
 		secrets.Insert(o.prometheusPassword)
 	}
+	if o.maxConcurrency == 0 {
+		o.maxConcurrency = runtime.GOMAXPROCS(0)
+	}
 	return nil
 }
 
+type clusterType int
+
+const (
+	none clusterType = iota
+	awsOnly
+	gcpOnly
+	moreThanOne
+
+	aws = "aws"
+	gcp = "gcp"
+)
+
+var (
+	knownCloudProviders = sets.NewString(aws, gcp)
+)
+
+func getClusterTypeFromEnv(spec *k8score.PodSpec) string {
+	if spec == nil {
+		return ""
+	}
+	for _, c := range spec.Containers {
+		for _, e := range c.Env {
+			if e.Name == "CLUSTER_TYPE" {
+				if knownCloudProviders.Has(e.Value) {
+					return e.Value
+				}
+			}
+		}
+	}
+	return ""
+}
+
+func getClusterType(jc *prowconfig.JobConfig, path string, config *dispatcher.Config) clusterType {
```

Comment:
Can we rename this to `getCloudproviderForCluster`? Its not really a type and will create confusion down the line, e.g. ppl use `clusterype` to refer to UPI vs IPI

Reply:
The semantics of this function are confusing, it returns the aggregated cloud provider of all jobs passed in?
Please explain at least in godoc what jobs you expect to be in the `jc` variable. One job only? All jobs for a given repo? All jobs we have (doesn't seem to make sense)?

Reply:
renamed to `getCloudProvidersForE2ETests ` and add gdoc for the function.